src/main/java/org/broadinstitute/hellbender/utils/bwa/Alignment.java

```diff
@@ -0,0 +1,142 @@
+    private final int suboptimalScore; // for bwa-specific XS tag
+    private final Cigar cigar;     // cigar for alignment (empty if unmapped)
+    private final String mdTag;    // the MD tag
+    private final String xaTag;    // the XA tag
+    private final int mateRefId;   // mate's refId (-1 if unpaired or if mate unmapped)
+    private final int mateRefStart;// mate's reference start (-1 if unpaired or if mate unmapped)
+    private final int templateLen; // inferred template length (0 if unpaired, mate unmapped, or on different ref contigs)
+
+    public Alignment( final int samFlag, final int refId, final int refStart, final int refEnd,
+                      final int tigStart, final int tigEnd, final int mapQual,
+                      final int nMismatches, final int alignerScore, final int suboptimalScore,
+                      final Cigar cigar, final String mdTag, final String xaTag,
+                      final int mateRefId, final int mateRefStart, final int templateLen ) {
+        this.samFlag = samFlag;
+        this.refId = refId;
+        this.refStart = refStart;
+        this.refEnd = refEnd;
+        this.tigStart = tigStart;
+        this.tigEnd = tigEnd;
+        this.mapQual = mapQual;
+        this.nMismatches = nMismatches;
+        this.alignerScore = alignerScore;
+        this.suboptimalScore = suboptimalScore;
+        this.cigar = cigar;
+        this.mdTag = mdTag;
+        this.xaTag = xaTag;
+        this.mateRefId = mateRefId;
+        this.mateRefStart = mateRefStart;
+        this.templateLen = templateLen;
+    }
+
+    public int getSamFlag() { return samFlag; }
+    public int getRefId() { return refId; }
+    public int getRefStart() { return refStart; }
+    public int getRefEnd() { return refEnd; }
+    public int getTigStart() { return tigStart; }
+    public int getTigEnd() { return tigEnd; }
+    public int getMapQual() { return mapQual; }
+    public int getNMismatches() { return nMismatches; }
+    public int getAlignerScore() { return alignerScore; }
+    public int getSuboptimalScore() { return suboptimalScore; }
+    public Cigar getCigar() { return cigar; }
+    public String getMDTag() { return mdTag; }
+    public int getMateRefId() { return mateRefId; }
+    public int getMateRefStart() { return mateRefStart; }
+    public int getTemplateLen() { return templateLen; }
+
+    public GATKRead apply( final GATKRead originalRead, final List<String> refNames, final SAMFileHeader header,
+                           final boolean softclipAlts, final boolean justLikeBWA ) {
+        final SAMRecord samRecord;
+        try {
+            samRecord = (SAMRecord)originalRead.convertToSAMRecord(header).clone();
+        } catch ( CloneNotSupportedException x ) {
+            throw new GATKException("can't clone SAMRecord", x);
+        }
+        samRecord.setFlags(samFlag);
+        if ( refId >= 0 ) samRecord.setReferenceName(refNames.get(refId));
```

Comment:
This may be getting beyond the scope of this PR, but there is a standard set of SAM tags [here](https://github.com/samtools/hts-specs/blob/master/SAMtags.pdf). Maybe we should sort out which tags are alignment-specific and provide a standard set of definitions programmatically in the engine somewhere (e.g. the SAMRecord adapter class). Thoughts?